File: app/src/main/java/com/example/roomdatabase/MyDao.java
package com.example.roomdatabase;

import androidx.room.*;

import java.util.List;


//second class
@Dao
public interface MyDao {


    //insert data in DB
    @Insert
    public void addUser(User user);

    //get Data from DB
    @Query("select * from users")
    public List<User> getUsers();

    //delete user from DB
    @Delete
    public void deleteUser(User user);

    //update user
    @Update
    public void updateUser(User user);
}

File: app/src/main/java/com/example/roomdatabase/MyDao.kt
package com.example.roomdatabase

import androidx.room.*


//second class
@Dao
interface MyDao {

    //get Data from DB
    @get:Query("select * from usersDB")
    val user: List<User>


    //insert data in DB
    @Insert
    fun addUser(user: User)

    //delete user from DB
    @Delete
    fun deleteUser(user: User)

    //update user
    @Update
    fun updateUser(user: User)
}
